independent-projects/extension-maven-plugin/src/main/java/io/quarkus/maven/ExtensionDescriptorMojo.java

```diff
@@ -678,17 +679,40 @@ public boolean visitLeave(DependencyNode node) {
     }
 
     private void addSource(ObjectNode extObject) throws MojoExecutionException {
-        Map<String, String> repo = ScmInfoProvider.getSourceRepo();
+        Scm scm = getScm();
+        String scmUrl = scm != null ? scm.getUrl() : null;
+
+        Map<String, String> repo = new ScmInfoProvider().getSourceRepo(scmUrl);
         if (repo != null) {
             ObjectNode metadata = getMetadataNode(extObject);
             for (Map.Entry<String, String> e : repo.entrySet()) {
-                // Tools may not be able to handle nesting in metadata, so do fake-nesting
-                metadata.put("scm-" + e.getKey(), e.getValue());
+                // Ignore if already set
+                String value = e.getValue();
+                String fieldName = "scm-" + e.getKey();
+                if (!metadata.has(fieldName) && value != null) {
+                    // Tools may not be able to handle nesting in metadata, so do fake-nesting
+                    metadata.put(fieldName, value);
+                }
 
             }
         }
     }
 
+    private Scm getScm() {
+        // We have three ways to do this; project.getScm() will query the derived model. Sadly, inherited <scm> entries are usually wrong, unless the parent is in the same project
+        // We can use getOriginalModel and getParent to walk the tree, but this will miss parents in poms outside the current execution, which might include a local reactor that we'd actually want to query
+        // Or we can use the bootstrap provider
+        Scm scm = null;
+        final Artifact artifact = project.getArtifact();
+        LocalProject localProject = workspaceProvider.getProject(artifact.getGroupId(), artifact.getArtifactId());
+        while (scm == null && localProject != null) {
+            scm = localProject.getRawModel().getScm();
+            localProject = localProject.getLocalParent();
+        }
```

Comment:
So you decided not to fallback to `MavenProject` API? Which is probably OK for a start. Perhaps we should log a warning though when we expected to locate the SCM metadata but failed.